python/lsst/analysis/tools/interfaces/_stages.py

```diff
@@ -182,13 +185,23 @@ def __call__(self, data: KeyedData, **kwargs) -> KeyedData:
         return results
 
 
+def _newNameChecker(value: str) -> bool:
+    if "-" in value:
+        # Yes this should be a log here, as pex config give us no other
+        # useful way to get info into the exception.
```

Comment:
Rather than a raise?

Typo: `give` to `gives`.

Reply:
Yeah, because I want the pex_execution framework to raise a `FieldValidationError` which it will do with the check function. That exception is the one that various tools will be looking for, but there also not an easy way to raise it yourself (you need access to various objects) I thought about hacking various complicated things to make it do what you might expect, but I figured it was not worth the time. This will log an exact message, and the standard exception WILL have  a message containing the actual text of the invalid value.